handbook/openid-connect/openid-connect.md

```diff
@@ -0,0 +1,133 @@
+
+#### The SystemLink login redirect URL
+```
+[protocol]://[systemlink-dns]/login/openidc-redirect
+
+```
+
+#### Provider Client Configuration Documentation
+
+[PingFederate - Configure an OAuth client](https://docs.pingidentity.com/bundle/pingfederate-93/page/roj1564002966901.html)
+
+[Google - Using OAuth 2.0 for Web Server Applications](https://developers.google.com/identity/protocols/oauth2/web-server)
+
+[Okta - Understand the callback route](https://developer.okta.com/docs/guides/sign-into-web-app/springboot/define-callback/)
+
+## Mapping OpenID Connect Claims to SystemLink Workspaces and Roles
+
+Once SystemLink and the provider have been configured such that users can authenticate and login, mappings between OpenID claims and workspaces must be made in order to provide the user access to systems and data managed by SystemLink. This process is the same as the mapping workflow for LDAP and Active Directory attributes, groups, and users. Please see [Assigning Users to Roles in a Workspace](https://www.ni.com/documentation/en/systemlink/latest/setup/mapping-roles/) in the SystemLink manual. Claims may also be used to create a mapping to the **Server Adminstrator** role. 
+
+The OpenID Connect provider determines which scopes and claims are made available to clients. The available claims can be viewed at the `userinfo_endpoint` hosted by the provider. Use `https://[provider-dns]/.well-known/openid-configuration` to determine the URL of the `userinfo_endpoint`. You will need to obtain a valid bearer token to authenticate and access this endpoint.  
+
+```bash
+curl -s https://slsso-runtime.grl-us1.uat.k8s.com/idp/userinfo.openid -H 'Authorization: Bearer eyJhbGciOiJSUzI1NiI...zJVy2oLnrBmXTmpDRm499U4~'|python -m json.tool
+```
+Example curl request to return user info. The bearer token has been shortened for readability. 
+
+```json
+{
+    "email": "mark.black@ni.com",
+    "family_name": "Black",
+    "given_name": "Mark",
+    "name": "Black",
+    "ni_employee": "2670",
+    "sub": "mblack"
+}
+```
+
+Example response from `userinfo_endpoint`. Any of these claims may be used to map a user to a role in a workspace. 
+
+![Mapping the ni_employee claim to a workspace](claim-mapping.png)
+*Mapping the `ni_employee` claim to a workspace*
+
+If a claim returns an array of values any one of these values can be supplied in the **Value** field to create a mapping. 
+
+## Troubleshooting Failed Authentication
+Due to the amount of configuration required by both the provider and SystemLink its can be challenging to enable the entire OpenID Connect flow. The following sources can be used to troubleshoot a failed connection. 
+
+**OpenID Connect Provider logs:** Consult your OpenID Connect Provider's documentation on the location of their application log files. 
+
+**NI Web Server Logs:** These are found at `C:\ProgramData\National Instruments\Web Server\logs\error.log`
```

Comment:
Related to the email Chris sent out yesterday, it might be worth clarifying that the suffixed version of this file oftentimes contains the latest logs. Users familiar with how Linux log files work might find this confusing.

Reply:
I checked into this and the log without the prefix was always the latest. This feels like a bug if that is not the case. Especially if it doesn't seem like the behavior is always the same.

Reply:
I saw it using files with a suffix yesterday when I sent the email.  @spanglerco had mentioned that it's the way the apache rotate log tool works on windows.

Reply:
My recommendation is to file a bug and resolve this issue rather than try and document around it here.